pkg/neonvm/ipam/ipam.go

```diff
@@ -82,9 +111,12 @@ func New(ctx context.Context, nadName string, nadNamespace string) (*IPAM, error
 	if err != nil {
 		return nil, fmt.Errorf("error creating kubernetes client: %w", err)
 	}
+	return NewWithClient(kClient, nadName, nadNamespace)
+}
 
+func NewWithClient(kClient *Client, nadName string, nadNamespace string) (*IPAM, error) {
 	// read network-attachment-definition from Kubernetes
-	nad, err := kClient.nadClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(nadNamespace).Get(ctx, nadName, metav1.GetOptions{})
+	nad, err := kClient.NADClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(nadNamespace).Get(context.Background(), nadName, metav1.GetOptions{})
```

Comment:
suggestion: We should not make a network call without a timeout, otherwise this can potentially block forever